### PR TITLE
Resolved cycle dependency in framework plugins registration

### PR DESCRIPTION
**Pull request by ahg-g**
Resolved cycle dependency in framework plugins registration and added a noop plugin to make sure we have the dependencies worked out.

**What type of PR is this?**
/kind bug

**Does this PR introduce a user-facing change?**:
```release-note
None
```

/sig scheduling

### CODE REVIEW

File: pkg/scheduler/framework/v1alpha1/registry.go [hunk 1]
```diff
@@ -68,19 +68,3 @@ func (r Registry) Unregister(name string) error {
 	delete(r, name)
 	return nil
 }
-
-// NewRegistry builds a default registry with all the default plugins.
-// This is the registry that Kubernetes default scheduler uses. A scheduler that
-// runs custom plugins, can pass a different Registry and when initializing the
-// scheduler.
-func NewRegistry() Registry {
```
Comment (hex108) [inline pkg/scheduler/framework/v1alpha1/registry.go]: A lot of places use it.

Comment (ahg-g) [reply pkg/scheduler/framework/v1alpha1/registry.go]: yeah, I will see how the various components fail and updated them.

**Review by draveness — COMMENTED** [review]
Is https://github.com/kubernetes/kubernetes/pull/80292 trying to solve the same problem?

**Comment by ahg-g** [comment]
> Is #80292 trying to solve the same problem?

yes, I spoke with @hex108 offline, this one will supersede #80292

**Review by Huang-Wei — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**